packages/shared/utils.ts

```diff
@@ -23,9 +23,35 @@ export function toNumber(value: string): number | string {
   return isNaN(n) ? value : n;
 }
 
+export function isObjectLike(value: any) {
+  return typeof value === 'object' && value !== null;
+}
+
+export function getTag(value: any) {
+  if (value == null) {
+    return value === undefined ? '[object Undefined]' : '[object Null]';
+  }
+  return Object.prototype.toString.call(value);
+}
+
+export function isPlainObject(value: any) {
+  // eslint-disable-next-line eqeqeq
+  if (!isObjectLike(value) || getTag(value) != '[object Object]') {
```

Comment:
Any specific reason why we use `!=` instead of `!==`. I would prefer we use `!==` instead of silencing the eslint rule.

Reply:
Maybe to install lodash and simply reuse `_.isPlainObject` https://lodash.com/docs/4.17.15#isPlainObject?

Because otherwise we will have to support and cover with tests in our codebase, what already covered in lodash

Reply:
I usually don't like including libs inside vee-validate due to bundle size. If it isn't far off (using Lodash.isPlainObject vs building our own) bundle-size wise (you can test it with `pnpm build`), Then happy to include it.

Reply:
Another option is to use [deepmerge](https://github.com/TehShrike/deepmerge) which is around 723B only and avoid having our own merge logic.

Reply:
I've tested bundle size and have following result:
Main branch size:
This is a base size
```sh
Output File: vee-validate.esm.js | Size: 123.59 kB | Gzip: 25.51 kB
Output File: vee-validate.js | Size: 123.07 kB | Gzip: 23.16 kB
Output File: vee-validate.min.js | Size: 40.04 kB | Gzip: 11.96 kB
```
---
Case 1. Without lodash deps, only copy-paste sources
```sh
Output File: vee-validate.esm.js | Size: 124.29 kB | Gzip: 25.65 kB
Output File: vee-validate.js | Size: 123.87 kB | Gzip: 23.3 kB
Output File: vee-validate.min.js | Size: 40.4 kB | Gzip: 12.04 kB
```
---
Case 2. With lodash deps
```sh
Output File: vee-validate.esm.js | Size: 129.37 kB | Gzip: 26.97 kB
Output File: vee-validate.js | Size: 119.8 kB | Gzip: 24.25 kB
Output File: vee-validate.min.js | Size: 41.21 kB | Gzip: 12.25 kB
```
---
Case 3. deepmerge + is-plain-object
By default, deepmerge doesn't have isPlainObject guard, so they recommend install additional deps https://github.com/TehShrike/deepmerge#ismergeableobject
I suppose it's the worst one
```sh
Output File: vee-validate.esm.js | Size: 128.54 kB | Gzip: 26.9 kB
Output File: vee-validate.js | Size: 118.93 kB | Gzip: 24.19 kB
Output File: vee-validate.min.js | Size: 41.88 kB | Gzip: 12.52 kB
```

What case may i implement?

Reply:
One more case with `lodash-es`
```
Output File: vee-validate.esm.js | Size: 128.95 kB | Gzip: 26.89 kB
Output File: vee-validate.js | Size: 122.63 kB | Gzip: 24.28 kB
Output File: vee-validate.min.js | Size: 41.05 kB | Gzip: 12.24 kB
```

Reply:
Let's go with case 1, smallest is always the best.

Reply:
Implemented and added some tests for coverage ✅